dockstore-cli-integration-testing/src/test/java/io/dockstore/client/cli/SingularityIT.java

```diff
@@ -0,0 +1,140 @@
+        WorkflowsApi workflowApi = new WorkflowsApi(webClient);
+        Workflow workflow = workflowApi.manualRegister(SourceControl.GITHUB.getFriendlyName(), "DockstoreTestUser2/md5sum-checker",
+                "/checker-workflow-wrapping-workflow.wdl", "test", "wdl", null);
+        workflowApi.refresh(workflow.getId());
+
+        // make a tmp copy of the dockstore config and add to it the cromwell conf file option
+        File tmpConfig = generateCromwellConfig();  // this is done in the test because the location varies
+
+        // start saving the output so as to make sure that Singularity actually runs instead of Docker
+        PrintStream old = System.out;  // save the regular output stream so it can be reset later
+        TeeOutputStream teeOut = new TeeOutputStream(System.out, outContent);
+        PrintStream out = new PrintStream(teeOut, true);  // also print it to the screen
+        System.setOut(out);
+
+        // run the md5sum-checker workflow
+        Client.main(new String[] {
+            "--config",
+            tmpConfig.getAbsolutePath(),
+            "workflow",
+            "launch",
+            "--entry",
+            SourceControl.GITHUB.toString() + "/DockstoreTestUser2/md5sum-checker/test",
+            "--json", ResourceHelpers.resourceFilePath("md5sum_wdl.json")
+        });
+        System.setOut(old);  // reset the output stream to stdout
+
+        // the phrase "singularity exec" will only be in the output if Singularity is actually being used
+        Assert.assertTrue(outContent.toString().contains("singularity exec"));
+    }
+
+    /**
+     * Makes a tmp file that's a copy of the contents of the config_for_singularity file in resources,
+     * and adds a line setting the cromwell-vm-options to the cromwell_singularity.conf path
+     * @return File object of the tmp file created
+     */
+    private File generateCromwellConfig() {
+        // get the path to the cromwell conf file in the current file system
+        // this configures cromwell to run with singularity instead of docker
+        String cromwellConfPath = ResourceHelpers.resourceFilePath("cromwell_singularity.conf");
+
+        try {
+            // make a new file in the tmp folder for this test
+            File tmpFile = temporaryFolder.newFile("config_for_singularity");
+            // copy the dockstore config for singularity from the resources directory into the new tmp file
+            File configTemplate = new File(ResourceHelpers.resourceFilePath("config_for_singularity"));
+            FileUtils.copyFile(configTemplate, tmpFile);
+
+            // append the new cromwell option to the tmp dockstore config
+            FileUtils.writeStringToFile(tmpFile, "\ncromwell-vm-options: -Dconfig.file=" + cromwellConfPath,
+                    StandardCharsets.UTF_8, true);
+            return tmpFile;
+
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
In tests, you usually don't catch exceptions 
(i.e. we want the test to fail if there are problems)

Reply:
ah ok. I guess I was thinking that an IOException from it failing to create or write to the file isn't the kind of error that's relevant to the test. But it would just fail later if the file isn't created, so I'll remove the catch block